diesel_codegen/src/schema_inference.rs

```diff
@@ -35,20 +35,20 @@ pub fn derive_infer_table_from_schema(input: syn::MacroInput) -> quote::Tokens {
     }
 
     let options = get_options_from_input(&input.attrs, bug).unwrap_or_else(|| bug());
-    let database_url = get_option(options, "database_url", bug);
-    let table_name = get_option(options, "table_name", bug);
+    let database_url = get_option(&options, "database_url", bug);
+    let table_name = get_option(&options, "table_name", bug);
 
-    let connection = establish_connection(database_url).unwrap();
-    let data = get_table_data(&connection, table_name).unwrap();
-    let primary_keys = get_primary_keys(&connection, table_name).unwrap()
+    let connection = establish_connection(&database_url).unwrap();
+    let data = get_table_data(&connection, &table_name).unwrap();
+    let primary_keys = get_primary_keys(&connection, &table_name).unwrap()
```

Comment:
Just for a fun exercise: I'm pretty sure the `&`s you added here are no longer necessary – the bindings are already references. (Deref coercion allows rustc to accept `&&&foo` where `&foo` was wanted.)